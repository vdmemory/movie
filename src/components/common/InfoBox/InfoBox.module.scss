.infoBox {
    border-radius: 6px;
    box-shadow: 0 0 0.2em rgba(0, 0, 0, .5);
    & .solid {
        height: 1px;
        background-color: #d3d2d2;
        width: 97%;
        margin: 0 auto;
    }
    & .title{
        font-size: 22px;
        line-height: 22px;
        padding: 15px 30px 15px;
        margin: 0;
    }
    & .content{
        & .leftBar{
            & img{}
        }
        & .rightBar{
           
        }
    }
    & .description{}
}