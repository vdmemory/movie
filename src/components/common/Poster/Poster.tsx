import React from "react";

import noposter from "./noposter.png";

type Props = {
    title: string;
    poster: string;
};

const Card: React.FC<Props> = ({ title, poster }) => {
    return <img alt={title} src={poster !== "N/A" ? poster : noposter} />;
};

export default Card;
