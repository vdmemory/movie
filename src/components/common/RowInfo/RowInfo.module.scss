.rowInfo {
    display: flex;
    align-items: center;
    padding: 5px 0;
    & .title {
        margin: 0;
        font-size: 20px;
    }

    & .subtitle {
        margin: 0 0 0 15px;
        font-size: 18px;
    }
}