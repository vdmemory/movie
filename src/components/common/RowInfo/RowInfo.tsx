import React from "react";

import styles from "./RowInfo.module.scss";

type Props = {
    title: string;
    subtitle: string;
};

const RowInfo: React.FC<Props> = ({ title, subtitle }) => {
    return (
        <div className={styles.rowInfo}>
            <h3 className={styles.title}>{title}</h3>
            <p className={styles.subtitle}>{subtitle}</p>
        </div>
    );
};

export default RowInfo;
