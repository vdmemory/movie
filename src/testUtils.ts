export const deepEqual = (a: any, b: any) => {
    if (a === b) {
        return true;
    }

    if (
        a == null ||
        typeof a != "object" ||
        b == null ||
        typeof b != "object"
    ) {
        return false;
    }

    var propertiesInA = 0,
        propertiesInB = 0;
    for (var property in a) {
        propertiesInA += 1;
    }
    for (var property in b) {
        propertiesInB += 1;
        if (!(property in a) || !deepEqual(a[property], b[property])) {
            return false;
        }
    }
    return propertiesInA == propertiesInB;
};
