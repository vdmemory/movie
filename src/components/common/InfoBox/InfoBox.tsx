import React from "react";

import RowInfo from "../RowInfo/RowInfo";

import styles from "./InfoBox.module.scss";

type Props = {
    title: string;
    year: string;
    runtime: string;
    genre: string;
    director: string;
    actors: string;
    plot: string;
    language: string;
    country: string;
    poster: string;
    imdbRating: string;
};

type RowInfo = {
    title: string;
    subtitle: string;
};

const InfoBox: React.FC<Props> = ({
    title,
    year,
    poster,
    genre,
    country,
    runtime,
    director,
    actors,
    language,
    plot,
}) => {
    const infoData = [
        { title: "Original title", subtitle: title },
        { title: "Genre", subtitle: genre },
        { title: "Country", subtitle: country },
        { title: "Year", subtitle: year },
        { title: "Duration", subtitle: runtime },
        { title: "Director", subtitle: director },
        { title: "Ð¡ast of actors", subtitle: actors },
        { title: "Language", subtitle: language },
    ];

    // const renderRowInfo = ({title, subtitle}) => {
    //     return <RowInfo title={title} subtitle={subtitle} />;
    // };

    return (
        <div className={styles.infoBox}>
            <h2 className={styles.title}>{`${title} (${year})`}</h2>
            <div className={styles.solid}></div>
            <div className={styles.content}>
                <div className={styles.leftBar}>
                    <img src={poster} alt={title} />
                </div>
                <div className={styles.rightBar}>
                    {/* {infoData.map(renderRowInfo)} */}
                    right bar
                </div>
            </div>
            <div className={styles.solid}></div>
            <p className={styles.description}>{plot}</p>
        </div>
    );
};

export default InfoBox;
