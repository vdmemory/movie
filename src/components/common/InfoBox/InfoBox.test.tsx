import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import StarRatingComponent from "react-star-rating-component";
import { BsStar } from "react-icons/bs";

import { deepEqual } from "../../../testUtils";
import InfoBox from "./InfoBox";
import RowInfo from "../RowInfo/RowInfo";
import Poster from "../Poster/Poster";

type Props = {
    title: string;
    year: string;
    runtime: string;
    genre: string;
    director: string;
    actors: string;
    plot: string;
    language: string;
    country: string;
    poster: string;
    imdbRating: string;
};

type RowInfo = {
    title: string;
    subtitle: string;
};

const setup = (props: Props) => {
    const renderer = createRenderer();
    renderer.render(<InfoBox {...props} />);
    const output = renderer.getRenderOutput();
    return { output, props, renderer };
};

const propsRender: Props = {
    title: "this title",
    year: "this year",
    runtime: "this runtime",
    genre: "this genre",
    director: "this director",
    actors: "this actors",
    plot: "this plot",
    language: "this language",
    country: "this country",
    poster: "this poster",
    imdbRating: "this imdbRating",
};

const infoData: { title: string; subtitle: string }[] = [
    { title: "Original title", subtitle: "this title" },
    { title: "Genre", subtitle: "this genge" },
];

describe("InfoBox Component", () => {
    describe("Render component and its components", () => {
        test("Should renders correctly", () => {
            const { output } = setup(propsRender);
            expect(output).toMatchSnapshot();
        });

        test("Should render container and has a movie class", () => {
            const { output } = setup(propsRender);
            expect(output.type).toBe("div");
            expect(output.props.className).toBe("infoBox");
        });

        test("Should render title and year and should have h2 tag", () => {
            const { output } = setup(propsRender);
            const [title] = output.props.children;
            expect(title.type).toBe("h2");
            expect(title.props.className).toBe("title");
            expect(title.props.children).toBe("this title (this year)");
        });

        describe("Block Content", () => {
            test("Should render description, check class", () => {
                const { output } = setup(propsRender);
                const [, , , desc] = output.props.children;
                expect(desc.props.children[1].type).toBe("p");
                expect(desc.props.children[1].props.className).toBe(
                    "description"
                );
            });

            test("Should render description text", () => {
                const { output } = setup(propsRender);
                const [, , , desc] = output.props.children;
                expect(desc.props.children[1].props.children).toBe("this plot");
            });

            test("Should not render description text if plot props = N/A", () => {
                const plotNotRender: Props = {
                    title: "this title",
                    year: "this year",
                    runtime: "this runtime",
                    genre: "this genre",
                    director: "this director",
                    actors: "this actors",
                    plot: "N/A",
                    language: "this language",
                    country: "this country",
                    poster: "this poster",
                    imdbRating: "this imdbRating",
                };
                const { output } = setup(plotNotRender);
                const [, , , desc] = output.props.children;
                expect(desc).toBe(null);
            });
        });

        describe("Block Content", () => {
            test("Should render content, check class", () => {
                const { output } = setup(propsRender);
                const [, , content] = output.props.children;
                expect(content.type).toBe("div");
                expect(content.props.className).toBe("content");
            });

            describe("Left Bar", () => {
                test("Should render left bar, check class", () => {
                    const { output } = setup(propsRender);
                    const [, , content] = output.props.children;
                    const [leftBar] = content.props.children;
                    expect(leftBar.type).toBe("div");
                    expect(leftBar.props.className).toBe("leftBar");
                });

                describe("Poster", () => {
                    const expectedProps = {
                        title: "this title",
                        poster: "this poster",
                    };
                    test("Should render and check props", () => {
                        const { output } = setup(propsRender);
                        const [, , content] = output.props.children;
                        const [leftBar] = content.props.children;

                        expect(leftBar.props.children.type).toBe(Poster);
                        expect(
                            deepEqual(
                                leftBar.props.children.props,
                                expectedProps
                            )
                        ).toBeTruthy();
                    });
                });
            });

            describe("Right Bar", () => {
                test("Should render right bar, check class", () => {
                    const { output } = setup(propsRender);
                    const [, , content] = output.props.children;
                    const [, rightBar] = content.props.children;
                    expect(rightBar.type).toBe("div");
                    expect(rightBar.props.className).toBe("rightBar");
                });

                test("Should render title in right bar, check class", () => {
                    const { output } = setup(propsRender);
                    const [, , content] = output.props.children;
                    const [, rightBar] = content.props.children;
                    const [, title] = rightBar.props.children;
                    expect(title.type).toBe("h3");
                    expect(title.props.className).toBe("titleScore");
                    expect(title.props.children).toBe("Total score:");
                });

                describe("renderRowInfo", () => {
                    test("Should render", () => {
                        
                        const { output } = setup(propsRender);
                        const [, , content] = output.props.children;
                        const [, rightBar] = content.props.children;
                        const [row, ,] = rightBar.props.children;
                        row.forEach((item: any, i: number) => {
                            expect(item.type).toBe(RowInfo);
                        });
                    });
                });

                describe("StarRatingComponent", () => {
                    test("Should render", () => {
                        const { output } = setup(propsRender);
                        const [, , content] = output.props.children;
                        const [, rightBar] = content.props.children;
                        const [, , star] = rightBar.props.children;
                        expect(star.type).toBe(StarRatingComponent);
                    });
                });
            });
        });
    });
});
