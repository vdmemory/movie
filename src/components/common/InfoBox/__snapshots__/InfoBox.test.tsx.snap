// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox Component Render component and its components Should renders correctly 1`] = `
<div
  className="infoBox"
>
  <h2
    className="title"
  >
    this title (this year)
  </h2>
  <div
    className="solid"
  />
  <div
    className="content"
  >
    <div
      className="leftBar"
    >
      <Card
        poster="this poster"
        title="this title"
      />
    </div>
    <div
      className="rightBar"
    >
      <RowInfo
        subtitle="this title"
        title="Original title"
      />
      <RowInfo
        subtitle="this genre"
        title="Genre"
      />
      <RowInfo
        subtitle="this country"
        title="Country"
      />
      <RowInfo
        subtitle="this year"
        title="Year"
      />
      <RowInfo
        subtitle="this runtime"
        title="Duration"
      />
      <RowInfo
        subtitle="this director"
        title="Director"
      />
      <RowInfo
        subtitle="this actors"
        title="Ð¡ast of actors"
      />
      <RowInfo
        subtitle="this language"
        title="Language"
      />
      <h3
        className="titleScore"
      >
        Total score:
      </h3>
      <StarRatingComponent
        editing={true}
        emptyStarColor="#333"
        name="imdbRating"
        renderStarIcon={[Function]}
        starColor="#ffb400"
        starCount={10}
        value={NaN}
      />
    </div>
  </div>
  <React.Fragment>
    <div
      className="solid"
    />
    <p
      className="description"
    >
      this plot
    </p>
  </React.Fragment>
</div>
`;
